api/v1/helpers.go

```diff
@@ -421,7 +410,13 @@ func (v *VerticaDB) GetUpgradePolicyToUse() UpgradePolicyType {
 	// the Auto option will automatically select this method, we first need to
 	// complete the implementation of this new policy.
 	if v.Spec.UpgradePolicy == ReplicatedUpgrade {
-		if v.HasSecondarySubclusters() && vinf.IsEqualOrNewer(ReplicatedUpgradeVersion) {
+		// Replicated upgrade requires that we scale out the cluster. See if
+		// there is evidence that we have already scaled past 3 nodes (CE
+		// license limit), or we have a license defined.
+		const ceLicenseLimit = 3
+		if vinf.IsEqualOrNewer(ReplicatedUpgradeVersion) &&
+			!v.IsKSafety0() &&
+			(v.getNumberOfNodes() > ceLicenseLimit || v.Spec.LicenseSecret != "") {
```

Comment:
It can happen if you install the license with other means. One example is a migrated database. You installed the license prior to moving to k8s. When you do migrate to k8s, you don't need to fill in the license secret since its already in the catalog.

Reply:
Yes that's correct, this makes sense! Thanks!